src/Microsoft.AspNetCore.ResponseCaching/Interfaces/IResponseCachingCacheKeyModifier.cs

```diff
@@ -5,13 +5,20 @@
 
 namespace Microsoft.AspNetCore.ResponseCaching
 {
-    public interface IResponseCachingCacheKeySuffixProvider
+    public interface IResponseCachingCacheKeyModifier
     {
         /// <summary>
+        /// Create a key segment that is prepended to the default cache key.
+        /// </summary>
+        /// <param name="httpContext">The <see cref="HttpContext"/>.</param>
+        /// <returns>The key segment that will be prepended to the default cache key.</returns>
+        string CreatKeyPrefix(HttpContext httpContext);
```

Comment:
My view is that we need at least a prefix, and the prefix is not hashed or part of the hash.
A simple configuration option at the middleware level would be fine. Even with multi-tenancy where each tenant would have its own middleware instance.